docs/design/fabric-v2+/fpc-registration.puml

```diff
@@ -7,266 +7,198 @@
+    (i.e., the verification report) is retrieved from untrusted environment.
+  end note
+  Enclave1 -> IAS ++: Attestation
+  return Evidence <- Verification_Report
+end group
 
-  |||
+  Enclave1 -> Enclave1 : Sealed_Credentials <- seal(CC_Parameters, Private_Keys)
 
-  Manager1 -> storage1 ++ : store sealed state
-  note right
-    sealed state is stored under a
-    unique identifier consisting of
-    CC_NAME, CC_VER, CC_ID, ECC_S_PK, ECC_E_PK
+  note over Enclave1
+    The constant string "CREATE" in the Verb field is meant to be used only here
+    (in the create call). A different constant string is expected to be used whenever
+    the Credentials prototobuf is reused (e.g., to re-attest the chaincode). This
+    allows to distinguish different calls. 
   end note
-  return
+  Enclave1 -> Enclave1 : Credentials <- <Verb="CREATE", Attested_Data, Evidence>
+  return Credentials, Sealed_Credentials
 
-  create entity IAS order 22
-  Manager1 -> IAS ++: send quote
-  note right
-    use IAS credentials and url provided by config
-  end note
-  note over Manager1 #FFaa00
-    **MST**: From an attestation generalization perspective i think there should be just a single
-    call ""GetAttestationProof"" or alike rather than a separate place where you get quote
-    and another where you get the registration report which requires some re-organization of calls.
-    A few observations on this:
-    - we do not strictly have to call the enclave via ECC. ECCManager could also directly call
-      enclave.so. this would make this creation and key-management potentially easier?
-    - the quote generation is not driven from within the enclave as currently shown above but from
-      the outside!
-    I think from a high-level there should be following three high-level calls either in
-    ECC or ECCMAnager depending on above:
-    - EnclaveCreate (key-gen & imprint of context, i.e., current create/binding flows
-    - EnclaveProof: gets org endorsement-sig, creates report, then quote
-      and, if epid, gets IAS report. 
-    - EnclavePersist: returns sealed state (important for security that this is called only after
-      EnclaveProof [which we might want to enforce using some state-machine?] )
-  end note
-  return ATTESTATION
+  return Credentials, Sealed_Credentials
 
-  |||
+  E_Manager -> E_Manager : store Sealed_Credentials
 
-  Manager1 -> Manager1 : mark ECC_ID as installed
-  note right
-    Manager1 maintains a list of
-    locally "installed/created" enclaves
-  end note
-  return ATTESTATION, ECC_S_PK / error
+  return result /' Admin -> E_Manager '/
 
   deactivate Admin
-  |||
 
 end
 
-group register attestation
+group Org-Enclave binding/certification (to be implemented)
+  Admin -> E_Manager ++: certify(CC_Id, Cert_Parameters)
+  E_Manager -> ECC1 ++: getCSR(Cert_Parameters)
+  ECC1 -> Enclave1 ++: getCSR(Cert_Parameters)
+  return csr /' ECC1 -> Enclave1 '/
```

Comment:
here some action is missing ... like `csr <- generate_csr(some inputs? cc_params???)`

Reply:
2285d1c0e8775289a66f534f16e8ba45b0f9d102